contracts/staking/DonationsStaking.sol

```diff
@@ -140,17 +132,32 @@ contract DonationsStaking is DAOUpgradeableContract {
 	 */
 	function setStakingContract(address _stakingContract) external {
 		_onlyAvatar();
-		(uint256 stakingAmount, ) =
-			stakingContract.getProductivity(address(this));
-		if (stakingAmount > 0)
-			stakingContract.withdrawStake(stakingAmount, false);
+		(uint256 stakingAmount, ) = stakingContract.getProductivity(address(this));
+		if (stakingAmount > 0) stakingContract.withdrawStake(stakingAmount, false);
 		uint256 stakingTokenBalance = stakingToken.balanceOf(address(this));
-		uint256 ethBalance = address(this).balance;
-		stakingToken.transfer(avatar, stakingTokenBalance);
-		address payable receiver = payable(avatar);
-		receiver.transfer(ethBalance);
+		uint256 safeAmount = IHasRouter(this).maxSafeTokenAmount(
+			address(stakingToken),
+			address(0x0),
+			stakingTokenBalance,
+			stakingContract.maxLiquidityPercentageSwap()
```

Comment:
the max percentage should be internal to the donationstaking it has nothing to do with stakingcontract

Reply:
....